File: pigeon/android_java/app/src/main/java/com/shinnaga/pigeon/FirstFragment.java
package com.shinnaga.pigeon;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Base64;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.shinnaga.pigeon.databinding.FragmentFirstBinding;

import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;
import io.flutter.embedding.engine.dart.DartExecutor;

import java.io.ByteArrayOutputStream;
import java.io.InputStream;

import com.shinnaga.pigeon.*;

public class FirstFragment extends Fragment {

    private FragmentFirstBinding binding;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {

        binding = FragmentFirstBinding.inflate(inflater, container, false);
        return binding.getRoot();

    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.buttonFirst.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                FlutterEngine flutterEngine = new FlutterEngine(view.getContext());
                String engineID = "param_engine";

                flutterEngine.getDartExecutor().executeDartEntrypoint(
                        DartExecutor.DartEntrypoint.createDefault()
                );

                FlutterEngineCache.getInstance()
                                .put(engineID, flutterEngine);

                Bitmap bitmap = BitmapFactory.decodeResource(getResources(), R.raw.usa);

                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
                bitmap.compress(Bitmap.CompressFormat.PNG, 100, byteArrayOutputStream);
                byte[] byteArray = byteArrayOutputStream.toByteArray();

                String encoded = Base64.encodeToString(byteArray, Base64.NO_WRAP);


                ParamApi.ParamColor paramColor = new ParamApi.ParamColor.Builder().setA(155L).setR(20L).setG(15L).setB(255L).build();

                ParamApi.Param param = new ParamApi.Param.Builder()
                        .setStr("test-java")
                        .setNum(100L)
                        .setColor(paramColor)
                        .setImage(encoded)
                        .build();

                new ParamApi.FlutterParamApi(flutterEngine.getDartExecutor().getBinaryMessenger()).setParams(param, unused -> {});

                startActivity(
                        FlutterActivity.withCachedEngine(engineID).build(view.getContext())
                );
                NavHostFragment.findNavController(FirstFragment.this)
                        .navigate(R.id.action_FirstFragment_to_SecondFragment);
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

}

File: pigeon/android_kotlin/app/src/main/java/com/shinnaga/pigeon/FirstFragment.kt
package com.shinnaga.pigeon

import android.net.Uri
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import com.shinnaga.pigeon.ParamApi.Param
import com.shinnaga.pigeon.databinding.FragmentFirstBinding
import io.flutter.embedding.android.FlutterActivity
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache
import io.flutter.embedding.engine.dart.DartExecutor
import android.util.Base64

/**
 * A simple [Fragment] subclass as the default destination in the navigation.
 */
class FirstFragment : Fragment() {

    private var _binding: FragmentFirstBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        _binding = FragmentFirstBinding.inflate(inflater, container, false)
        return binding.root

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        lateinit var flutterEngine : FlutterEngine
        var engineID = "param_engine"
        binding.buttonFirst.setOnClickListener {
            flutterEngine = FlutterEngine(requireActivity().applicationContext)

            flutterEngine.dartExecutor.executeDartEntrypoint(
                DartExecutor.DartEntrypoint.createDefault()
            )

            FlutterEngineCache
                .getInstance()
                .put(engineID, flutterEngine)

            val inputStream = context?.resources?.openRawResource(R.drawable.usa)
            val bytes = inputStream?.readBytes()
            val encoded = Base64.encodeToString(bytes, Base64.NO_WRAP)

            val param = Param.Builder()
                .setStr("test-kotlin")
                .setNum(0xFF000000)
                .setColor(ParamApi.ParamColor.Builder().setA(50).setR(0).setG(130).setB(220).build())
                .setImage(encoded)
                .setListStr(listOf(encoded, encoded, encoded, encoded, encoded, encoded, encoded, encoded, encoded, encoded, encoded, encoded))
                .build()

            ParamApi.FlutterParamApi(flutterEngine.dartExecutor.binaryMessenger).setParams(param) {
            }

            startActivity(
                FlutterActivity
                    .withCachedEngine(engineID)
                    .build(requireActivity().applicationContext)
            )
            findNavController().navigate(R.id.action_FirstFragment_to_SecondFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}